docs/basics/shell.rst

```diff
@@ -0,0 +1,52 @@
+.. _shell:
+
+Using Common Shell Commands on DataLad Repositories
+---------------------------------------------------
+
+There are a number of gotchas when using common shell commands to manipulate DataLad Repositories.
+This stems mostly due to the reliance of git-annex on symlinks, as well as due to permissions management under the ``.git`` directory.
+Thus, the usage of common shell tools, including `GNU coreutils <https://www.gnu.org/software/coreutils/>`_, can often yield unintuitive results.
+This is a list referencing such cases, providing instructions on how to mitigate them, as well as indicating alternative processes using datalad-specific commands, where such are feasible substitutes.
+
+Removal via ``rm``
+~~~~~~~~~~~~~~~~~~
+
+Problem
+^^^^^^^
+
+Removing datalad repositories with ``rm`` will fail due to missing write permissions under ``.git/annex/objects/..``:
+
+.. code-block::
+   [deco]/tmp ❱ datalad clone https://github.com/datalad/testrepo--basic--r1
+   [INFO   ] Remote origin not usable by git-annex; setting annex-ignore
+   [INFO   ] https://github.com/datalad/testrepo--basic--r1/config download failed: Not Found
+   install(ok): /tmp/testrepo--basic--r1 (dataset)
+   [deco]/tmp ❱ cd testrepo--basic--r1/
+   [deco]/tmp/testrepo--basic--r1 ❱ datalad get .
+   get(ok): test-annex.dat (file) [from web...]
+   [deco]/tmp/testrepo--basic--r1 ❱ cd ..
+   [deco]/tmp ❱ rm -rf testrepo--basic--r1/
+   rm: cannot remove 'testrepo--basic--r1/.git/annex/objects/zk/71/SHA256E-s4--181210f8f9c779c26da1d9b2075bde0127302ee0e3fca38c9a83f5b1dd8e5d3b.dat/SHA256E-s4--181210f8f9c779c26da1d9b2075bde0127302ee0e3fca38c9a83f5b1dd8e5d3b.dat': Permission denied
+
```

Comment:
The ``.. code-block::`` directives currently cause a build error because they need a space between the directive and the code:
```suggestion
.. code-block::

   [deco]/tmp ❱ datalad clone https://github.com/datalad/testrepo--basic--r1
   [INFO   ] Remote origin not usable by git-annex; setting annex-ignore
   [INFO   ] https://github.com/datalad/testrepo--basic--r1/config download failed: Not Found
   install(ok): /tmp/testrepo--basic--r1 (dataset)
   [deco]/tmp ❱ cd testrepo--basic--r1/
   [deco]/tmp/testrepo--basic--r1 ❱ datalad get .
   get(ok): test-annex.dat (file) [from web...]
   [deco]/tmp/testrepo--basic--r1 ❱ cd ..
   [deco]/tmp ❱ rm -rf testrepo--basic--r1/
   rm: cannot remove 'testrepo--basic--r1/.git/annex/objects/zk/71/SHA256E-s4--181210f8f9c779c26da1d9b2075bde0127302ee0e3fca38c9a83f5b1dd8e5d3b.dat/SHA256E-s4--181210f8f9c779c26da1d9b2075bde0127302ee0e3fca38c9a83f5b1dd8e5d3b.dat': Permission denied
```

However, what I would suggest we do is transforming them into a ``.. autorunrecord::`` directive. These get executed automatically by appveyor and will always show how the current software stack currently behaves, i.e., they are far better maintainable. 

```suggestion
.. runrecord:: _examples/DL-101-136-185
   :language: console
   :workdir: dl-101
   :exitcode: 1
   
   $ datalad clone https://github.com/datalad/testrepo--basic--r1
   $ cd testrepo--basic--r1/
   $ datalad get .
   $ cd ..
   $ rm -rf testrepo--basic--r1/
```

Any line starting with a ``$`` will be executed in sequence in the ``dl-101`` workdir (on Appveyor), and the output of all commands will be written to ``_examples/DL-101-136-185``. The code block expects a non-zero exit code (I hope 1 is the correct exit code, I didn't try). The [Contributing](https://handbook.datalad.org/en/latest/contributing.html) page has a dedicated section that explains the technical backbone behind it.

Reply:
@adswa I saw that in the other examples, the issue with that is the prompt doesn't show the current working directory, which makes things a bit unclear. Is there any way to do that?

Also:

> The code block expects a non-zero exit code 

given the nature of this section, a lot of the commands are supposed to fail, so “proper” examples of what fails might show up as errors.

Reply:
> @adswa I saw that in the other examples, the issue with that is the prompt doesn't show the current working directory, which makes things a bit unclear. Is there any way to do that?

In theory, we could configure it like this, but a cheap way to do it is to add a comment line (``#`` prefix) where you simply state "Outside of the dataset".  

> given the nature of this section, a lot of the commands are supposed to fail, so “proper” examples of what fails might show up as errors.

Sorry, I should have been more clear on this. Whenever a runrecord is meant to fail for whatever reason (we often show errors on purpose), it needs this annotation. Otherwise, the CI system treats it as a failure.